docs/DxcOnLinux.rst

```diff
@@ -0,0 +1,78 @@
+=======================================
+DirectXShaderCompiler on Linux & Mac OS
+=======================================
+
+.. contents::
+   :local:
+   :depth: 3
+
+Introduction
+============
+
```

Comment:
We should briefly explain why we are doing the port in case ppl know that DXC is based on LLVM/Clang and assumed it is naturally cross-platform. without knowing the internals:

DirectXShaderCompiler (DXC) is based on LLVM/Clang, which is originally cross-platform. However, to support HLSL, certain Windows specific techniques (like COM, SAL, etc.) are introduced to solve technical issues on the Windows platform, which also makes DXC not compilable/runnable on non-Windows platforms. Upon several requests ...

Reply:
Done